### PR TITLE
feat: CDC flow for ineligible transactions

### PR DESCRIPTION
**Pull request by dalsontws**
- [x] Screen for not eligible NRICs
- [x] unit tests

integration with mystique

- [x] throw EligibilityError for specific error message
- [x] catch EligibilityError and set cart state to Not Eligible

Note: 
Catch EligibilityError and set cart state is used ahead of checking the quotaResponse to determine error, as response is `null`, and could be due to other reasons

### CODE REVIEW

**Comment by ilhaoni** [comment]
Currently we haven’t integrate with mystique to try yet right haha

Should we update the useCart.test as well? Since we now have new states ((:

File: src/hooks/useCart/useCart.tsx [hunk 1]
```diff
@@ -131,8 +137,10 @@ export const useCart = (
         }
         setQuotaResponse(quotaResponse);
       } catch (e) {
-        // Cart will remain in FETCHING_QUOTA state.
-        if (e instanceof PolicyError) {
+        if (e instanceof EligibilityError) {
```
Comment (szenius) [inline src/hooks/useCart/useCart.tsx:140]: Nit, would it be more straight forward to call this `NotEligibleError`?

Comment (szenius) [inline src/hooks/useCart/useCart.tsx:140]: Nit, would it be more straight forward to call this NotEligibleError?

File: src/hooks/useCart/useCart.tsx [hunk 2]
```diff
@@ -131,8 +137,10 @@ export const useCart = (
...
+          setCartState("NOT_ELIGIBLE");
```
Comment (szenius) [inline src/hooks/useCart/useCart.tsx:141]: Need to write tests~~

File: src/hooks/useCart/useCart.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (dalsontws) [reply src/hooks/useCart/useCart.tsx:140]: yeah wont! haha the error message needs to be specific to throw eligibilityError hehe

Comment (dalsontws) [reply src/hooks/useCart/useCart.tsx:140]: agreed haha!!👌

File: src/hooks/useCart/useCart.test.tsx [hunk 3]
```diff
@@ -159,6 +164,13 @@ const mockPostTransactionResult = {
   ]
 };
 
+const mockQuotaNotEligible: any = (id: string) => {
```
Comment (szenius) [inline src/hooks/useCart/useCart.test.tsx:167]: Small nit too: might be more understandable if the quota word was replaced here, cuz it's not really that the quota that was not eligible

Comment (dalsontws) [reply src/hooks/useCart/useCart.test.tsx]: got it, changed! :)

File: src/components/CustomerQuota/IneligibleCard.tsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  return (
+    <View>
+      <CustomerCard nrics={nrics} headerBackgroundColor={color("red", 60)}>
+        <View
+          style={[
+            sharedStyles.resultWrapper,
+            sharedStyles.failureResultWrapper
+          ]}
+        >
+          <Text style={sharedStyles.emoji}>❌</Text>
```
[the remark this answers is not in the document]
Comment (dalsontws) [reply src/components/CustomerQuota/IneligibleCard.tsx]: agreee! adding this to new card, to discuss with Eida as well 👍

**Pull request merged**